webapp/src/Service/ScoreboardService.php

```diff
@@ -335,6 +340,34 @@ public function calculateScoreRow(
         $submissionsPubl = $pendingPubl = $timePubl = 0;
         $correctJury     = false;
         $correctPubl     = false;
+        $runtimeJury     = PHP_INT_MAX;
+        $runtimePubl     = PHP_INT_MAX;
+
+        // do runtime separately to not interfere with break/continue logic in normal submission counting
+        // we want to allow runtime improvements even after first correct submission
```

Comment:
I think it would be cleaner to not duplicate this loop, but instead make the `continue/break` logic depend on the scoring type. Creating code duplication is asking for bugs when we change something in one place and overlook the other.